plugins/Deflect/DeflectPlugin.cpp

```diff
@@ -81,31 +81,32 @@ class DeflectPlugin::Impl
                 _closeStream();
         }
 
-        if (_previousHost != _params.getHostname())
-        {
-            _params.setEnabled(true);
-            _previousHost = _params.getHostname();
-        }
-
         const bool deflectEnabled = _params.getEnabled();
```

Comment:
add !_params.getEnabled() to the if (_stream) statement.
Check for _params.getEnabled() here and if false _params.resetModified() and early return. You don't need the deflectEnabled variable afterwards.

Reply:
I'm lost. Apparently you are too from the current code. We should look at this in person.